### ISSUE TITLE
Pftools Python Run init API

### ISSUE THREAD

**Issue by psavery** [parflow/parflow#287]
In the pftools python examples, Run objects are currently being initialized like the following:
```python
drich = Run('default_richards', __file__)
```

However, we have the ability in Python to find the file that instantiated the object, which would remove the need for `__file__` argument. But we might want to discuss what the API should look like. I know Jupyter is a special case, and there are probably some others too.

What if we have 2 arguments for specifying the working directory, such as `working_dir` (defaults to `None`), and `default_to_script_working_dir` (defaults to `True`).

Basically, the user would be able to specify a working directory if they want to. If they don't, and `default_to_script_working_dir` is `True`, then the working dir will be the script directory (same as previously specifying `__file__`). If `working_dir` is `None` and `default_to_script_working_dir` is `False`, then the working directory won't be changed.

We might also be able to do something to check for special cases like running from the Python terminal or running from a Jupyter notebook, and adapt the working dir depending on where it is running from.

@reedmaxwell @lecondon thoughts?

**Comment by jourdain** [comment]
@reedmaxwell @lecondon
